### PR TITLE
fix not to add git commit link in about card edit button

### LINKED ISSUE #22140: 🐛 Bug Report: Edit button of the About Card leads the user to a non-existing page if the entity is loaded from a tag
### 📜 Description

Hello!
When the user has an entity loaded from a repository tag, the edit button in the About card leads the user to the page <repository-url>/edit/<tag>/catalog-info.yaml - which is an unexisting page for tags (this only works for branches).
edit-button-about-card

### 👍 Expected behavior

Since it's not possible to edit a tag, the button should be hidden. 

### 👎 Actual Behavior with Screenshots

The button leads the user to a not found page.
gh-404-page

### 👟 Reproduction steps

1. Register an entity from a tagged yaml file. 
2. Go to the entity page in Backstage.
3. Select the edit button in the About Card.

### 📃 Provide the context for the Bug.

This issue is leading users to a broken page, and affects user usability.

### 🖥️ Your Environment

OS: Linux 6.4.16-linuxkit - linux/x64 
node: v18.18.0 
yarn: 3.6.0 cli: 0.22.13 (installed) 
backstage: v1.18.4
Dependencies: @backstage/app-defaults 1.4.3 @backstage/backend-app-api 0.5.5 @backstage/backend-common 0.19.7 @backstage/backend-dev-utils 0.1.1 @backstage/backend-openapi-utils 0.0.4 @backstage/backend-plugin-api 0.6.5 @backstage/backend-tasks 0.5.10 @backstage/catalog-client 1.4.4 @backstage/catalog-model 1.4.2 @backstage/cli-common 0.1.12 @backstage/cli-node 0.1.4 @backstage/cli 0.22.13 @backstage/config-loader 1.5.0 @backstage/config 1.1.0 @backstage/core-app-api 1.10.0 @backstage/core-components 0.13.5 @backstage/core-plugin-api 1.6.0 @backstage/dev-utils 1.0.21 @backstage/errors 1.2.2 @backstage/eslint-plugin 0.1.3 @backstage/integration-aws-node 0.1.6 @backstage/integration-react 1.1.19 @backstage/integration 1.7.0 @backstage/plugin-api-docs 0.9.11 @backstage/plugin-auth-backend-module-gcp-iap-provider 0.1.2 @backstage/plugin-auth-backend-module-github-provider 0.1.2 @backstage/plugin-auth-backend-module-gitlab-provider 0.1.2 @backstage/plugin-auth-backend-module-google-provider 0.1.2 @backstage/plugin-auth-backend-module-oauth2-provider 0.1.2 @backstage/plugin-auth-backend 0.19.2 @backstage/plugin-auth-node 0.2.19, 0.3.2 @backstage/plugin-catalog-backend-module-github 0.4.3 @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.2 @backstage/plugin-catalog-backend-module-unprocessed 0.3.2 @backstage/plugin-catalog-backend 1.13.3 @backstage/plugin-catalog-common 1.0.16 @backstage/plugin-catalog-graph 0.2.36 @backstage/plugin-catalog-import 0.10.0 @backstage/plugin-catalog-node 1.4.6 @backstage/plugin-catalog-react 1.8.4 @backstage/plugin-catalog-unprocessed-entities 0.1.3 @backstage/plugin-catalog 1.13.0 @backstage/plugin-circleci 0.3.24 @backstage/plugin-entity-feedback-backend 0.2.2 @backstage/plugin-entity-feedback-common 0.1.3 @backstage/plugin-entity-feedback 0.2.7 @backstage/plugin-events-backend-module-github 0.1.15 @backstage/plugin-events-backend-test-utils 0.1.15 @backstage/plugin-events-backend 0.2.14 @backstage/plugin-events-node 0.2.14 @backstage/plugin-github-actions 0.6.5 @backstage/plugin-home-react 0.1.3 @backstage/plugin-home 0.5.8 @backstage/plugin-org 0.6.14 @backstage/plugin-permission-backend 0.5.28 @backstage/plugin-permission-common 0.7.8 @backstage/plugin-permission-node 0.7.16 @backstage/plugin-permission-react 0.4.15 @backstage/plugin-proxy-backend 0.4.2 @backstage/plugin-scaffolder-backend-module-cookiecutter 0.2.29 @backstage/plugin-scaffolder-backend 1.17.3 @backstage/plugin-scaffolder-common 1.4.1 @backstage/plugin-scaffolder-node 0.2.5 @backstage/plugin-scaffolder-react 1.5.5 @backstage/plugin-scaffolder 1.15.0 @backstage/plugin-search-backend-module-catalog 0.1.9 @backstage/plugin-search-backend-module-techdocs 0.1.9 @backstage/plugin-search-backend-node 1.2.9 @backstage/plugin-search-backend 1.4.5 @backstage/plugin-search-common 1.2.6 @backstage/plugin-search-react 1.7.0 @backstage/plugin-search 1.4.0 @backstage/plugin-sonarqube-backend 0.2.7 @backstage/plugin-sonarqube-react 0.1.8 @backstage/plugin-sonarqube 0.7.5 @backstage/plugin-techdocs-backend 1.7.2 @backstage/plugin-techdocs-node 1.8.2 @backstage/plugin-techdocs-react 1.1.11 @backstage/plugin-techdocs 1.7.0 @backstage/plugin-user-settings 0.7.10 @backstage/release-manifests 0.0.10 @backstage/test-utils 1.4.3 @backstage/theme 0.4.2 @backstage/types 1.1.1 @backstage/version-bridge 1.0.5

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by npiyush97**
## Hey, I just made a Pull Request!
closes #22140 

check if URL link is of commit hash. Better regex suggestion ?
Should address this to user ?
like if user passes invalid link in app-config.yaml just get ignored.User is not notified that URL is wrong in some way.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by npiyush97** [comment]
just to have a look @vinzscam

File: plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts [hunk 1]
```diff
@@ -53,12 +53,16 @@ export class AnnotateLocationEntityProcessor implements CatalogProcessor {
     let viewUrl;
     let editUrl;
     let sourceLocation;
+    const gitCommitBranchURLPattern = /\b[0-9a-f]{5,40}\b/;
```
Comment (vinzscam) [inline plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts:56]: unfortunately this will also match branches

**Review by vinzscam — CHANGES_REQUESTED** [review]

File: plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts (hunk 1 above)
Comment (npiyush97) [reply plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts:56]: yes right now testing this 
const commitHashRegex = /\/blob\/[0-9a-f]{40}\//;

**Comment by npiyush97** [comment]
n

File: plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts (hunk 1 above)
Comment (Rugvip) [reply plugins/catalog-backend/src/modules/core/AnnotateLocationEntityProcessor.ts:56]: I'm thinking `{40,}` could be good? It's match both sha1 and future formats. We won't even have commit hash shorthands in the URL right? At least doesn't seem like something we have to support

File: .changeset/thirty-plums-shout.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Added a regex test to check commit hash.If url is from git commit branch ignore the edit url.
```
Comment (Rugvip) [inline .changeset/thirty-plums-shout.md:5]:
```suggestion
Added a regex test to check commit hash. If url is from git commit branch ignore the edit url.
```

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍

**Comment by npiyush97** [comment]
ping @Rugvip

**Comment by npiyush97** [comment]
@Rugvip need this ?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Realized there's a tiny extra fix needed, but otherwise good to go

**Comment by npiyush97** [comment]
Ping again @Rugvip

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit: 

Thank you! 👍

**Pull request merged**